src/Microsoft.DotNet.XHarness.InstrumentationBase.Xunit/AndroidInstrumentationBase.cs

```diff
@@ -0,0 +1,198 @@
+                }
+
+                resultsPath = Path.Combine(docsDir, arguments[ResultsFileArgumentName]);
+                includeMethod = arguments[IncludeMethodArgumentName];
+                excludeCategories = arguments[ExcludeCategoriesArgumentName];
+                excludeMethod = arguments[ExcludeMethodArgumentName];
+                includeClass = arguments[IncludeClassArgumentName];
+                excludeClass = arguments[ExcludeClassArgumentName];
+            }
+
+            protected override bool LogExcludedTests => true;
+
+            public override TextWriter Logger => null;
+
+            public override string TestsResultsFinalPath => resultsPath;
+
+            protected override int? MaxParallelThreads => System.Environment.ProcessorCount;
+
+            protected override IDevice Device => null;
+
+            public IEnumerable<string> Tests { get; internal set; }
+
+            protected override IEnumerable<TestAssemblyInfo> GetTestAssemblies()
+            {
+                //yield return new TestAssemblyInfo(Assembly.GetExecutingAssembly(), Assembly.GetExecutingAssembly().Location);
+                //yield return new TestAssemblyInfo(typeof(Battery_Tests).Assembly, typeof(Battery_Tests).Assembly.Location);
+                foreach (string file in Tests)
+                {
+                    yield return new TestAssemblyInfo(Assembly.LoadFrom(file), file);
+                }
+            }
+            protected override void TerminateWithSuccess()
+            {
+            }
+
+            protected override TestRunner GetTestRunner(LogWriter logWriter)
+            {
+                var testRunner = base.GetTestRunner(logWriter);
+                if (excludeCategories != null)
+                {
+                    testRunner.SkipCategories(excludeCategories.Split(' '));
+                }
+                if (excludeMethod != null)
+                {
+                    foreach (var method in excludeMethod.Split(' '))
+                    {
+                        testRunner.SkipMethod(method, true);
+                    }
+                }
+                if (excludeClass != null)
+                {
+                    foreach (var item in excludeClass.Split(' '))
+                    {
+                        testRunner.SkipMethod(item, true);
+                    }
+                }
+                if (includeMethod != null)
+                {
+                    foreach (var method in includeMethod.Split(' '))
+                    {
+                        testRunner.SkipMethod(method, false);
+                    }
+                }
+                if (includeClass != null)
+                {
+                    foreach (var item in includeClass.Split(' '))
+                    {
+                        testRunner.SkipMethod(item, false);
+                    }
+                }
+                return testRunner;
+            }
```

Comment:
Just as an exercise, you do not need to write that many times the same code, you can see the pattern:
```csharp
if (filter != null) {
   foreach (var f in fiter.Split (' ') {
      Lambda (f, included);
   }
}
```
where Lambda:
```csharp
Lambda = Action<string, bool> ()
```
For ref: https://docs.microsoft.com/en-us/dotnet/api/system.action-2?view=net-5.0

So, you can get all those lines in a single method, and no need to copy paste which is error prone.